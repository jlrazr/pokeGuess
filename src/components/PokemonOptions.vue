<template>
  <div class="container">
    <ul>
      <li
        v-for="pokemon in pokemons"
        :key="pokemon.data.id"
        @click="disableOptions(), $emit('pickPokemon', pokemon.data.id)"
        :class="[isButtonActive ? 'enabled' : 'disabled']"
      >
        {{ pokemon.data.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    pokemons: {
      type: Array,
      required: true,
    },
  },

  data() {
    return {
      isButtonActive: {
        type: Boolean,
        default: true,
      },
    };
  },

  methods: {
    pickPokemon() {
      return;
    },

    disableOptions() {
      this.isButtonActive = false;
    },
  },
};
</script>

<style lang="scss" scoped>
.container {
  display: flex;
  justify-content: center;
  align-items: center;

  ul {
    list-style-type: none;

    li {
      margin-right: 50px;
      background: #fff;
      border-radius: 5px;
      border: 1px solid rgba(0, 0, 0, 0.2);
      cursor: pointer;
      margin-bottom: 10px;
      width: 250px;
      padding: 14px 5px;
      font-weight: 600;

      &:hover {
        color: #fff;
        background: rgba(0, 0, 0, 0.05);
      }
    }

    .disabled {
      pointer-events: none;
      color: darkgray;
      background: grey;
    }
  }
}
</style>
