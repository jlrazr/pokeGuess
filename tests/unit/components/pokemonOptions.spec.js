import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";
import { pokemons } from "../mocks/pokemons.mock";

describe("PokemonOptions Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, { props: { pokemons } });
  });

  test("must match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("must show the 4 pokemons options correctly", () => {
    const liTags = wrapper.findAll("li");
    expect(liTags.length).toBe(4);
    expect(liTags[0].text()).toBe("MewTwo");
    expect(liTags[1].text()).toBe("Bulbasaur");
    expect(liTags[2].text()).toBe("Ivysaur");
    expect(liTags[3].text()).toBe("Voltorb");
  });

  test("must emmit the selection with its own paramenters when clicked", () => {
    const [li1, li2, li3, li4 ] = wrapper.findAll("li");

    li1.trigger("click");
    li2.trigger("click");
    li3.trigger("click");
    li4.trigger("click");

    expect(wrapper.emitted("pickPokemon").length).toBe(4);
    expect(wrapper.emitted("pickPokemon")[0]).toStrictEqual([150]);
    expect(wrapper.emitted("pickPokemon")[1]).toStrictEqual([1]);
    expect(wrapper.emitted("pickPokemon")[2]).toStrictEqual([2]);
    expect(wrapper.emitted("pickPokemon")[3]).toStrictEqual([100]);
  });
});
