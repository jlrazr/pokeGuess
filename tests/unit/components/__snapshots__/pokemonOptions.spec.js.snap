// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component must match the snapshot 1`] = `
<div class="container">
  <ul>
    <li class="enabled">MewTwo</li>
    <li class="enabled">Bulbasaur</li>
    <li class="enabled">Ivysaur</li>
    <li class="enabled">Voltorb</li>
  </ul>
</div>
`;
